diff -Naur zipgateway/usr/local/beaglebone/install_bb.sh zipgateway_mod/usr/local/beaglebone/install_bb.sh
--- zipgateway/usr/local/beaglebone/install_bb.sh	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/beaglebone/install_bb.sh	2017-01-02 15:30:38.453899146 +0530
@@ -4,23 +4,29 @@
 	/etc/init.d/zipgateway stop
 	sleep 2
 fi
+mkdir -p /etc/zipgateway
+mkdir -p /etc/init.d/
 cp usr/bin/zipgateway /usr/local/bin/
 cp etc/zipgateway.tun /usr/local/etc/
 cp etc/zipgateway.cfg /usr/local/etc/
 cp etc/init.d/zipgateway /etc/init.d/
 cp etc/*.pem /usr/local/etc
-read -p "What is the serial port for zwave controller (default: /dev/ttyACM0): " controller
+read -p "What is the serial port for zwave controller (default: /dev/ttyS0): " controller
 if [ -z $controller ]
 then
-	echo "Using ZipSerialAPIPortName=/dev/ttyACM0"
+	echo "Using ZipSerialAPIPortName=/dev/ttyS1"
 	sed -i "/ZipSerialAPIPortName*/d" /usr/local/etc/zipgateway.cfg
-	echo ZipSerialAPIPortName=/dev/ttyACM0 >> /usr/local/etc/zipgateway.cfg
+	echo ZipSerialAPIPortName=/dev/ttyS1 >> /etc/zipgateway/zipgateway.cfg
 	echo "Note: If you want to change the port later change it in /usr/local/etc/zipgateway.cfg"
 else
 	sed -i "/ZipSerialAPIPortName*/d" /usr/local/etc/zipgateway.cfg
 	echo ZipSerialAPIPortName=$controller
 	echo ZipSerialAPIPortName=$controller >> /usr/local/etc/zipgateway.cfg
 fi
+sed -i "/ZipCaCert*/d" /usr/local/etc/zipgateway.cfg
+sed -i "/ZipCert*/d" /usr/local/etc/zipgateway.cfg
+sed -i "/ZipPrivKey*/d" /usr/local/zipgateway.cfg
+echo ZipPSK=12345678901234567890123456789012 >> /usr/local/etc/zipgateway.cfg
 update-rc.d zipgateway defaults 
 echo "Disabling connman"
 systemctl disable connman.service
diff -Naur zipgateway/usr/local/beaglebone/zipgateway.bb zipgateway_mod/usr/local/beaglebone/zipgateway.bb
--- zipgateway/usr/local/beaglebone/zipgateway.bb	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/beaglebone/zipgateway.bb	2017-01-02 15:33:05.769907160 +0530
@@ -14,15 +14,16 @@
 	echo "starting zipgateway"
 	ifconfig br-lan > /dev/null 2>&1 || {
 		brctl addbr br-lan
-		brctl addif br-lan eth0
 	}
-	ifconfig eth0 up
 	ifconfig br-lan up
-	${BINARY} -e /etc/eeprom.dat &
+	ifconfig br-lan 192.168.168.1
+	/usr/sbin/udhcpd /etc/zipgateway/udhcpd.conf
+	${BINARY} -e /etc/zipgateway/eeprom.dat &
 	;;
 stop)
 	echo "stopping zipgateway"
 	killall -SIGINT ${BINARY}
+	killall udhcpd
 	;;
 *)
 	echo "Usage: /etc/init.d/zipgateway {start|stop}"
diff -Naur zipgateway/usr/local/contiki/cpu/native/linux-serial.c zipgateway_mod/usr/local/contiki/cpu/native/linux-serial.c
--- zipgateway/usr/local/contiki/cpu/native/linux-serial.c	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/contiki/cpu/native/linux-serial.c	2017-01-02 15:33:48.585909489 +0530
@@ -4,7 +4,7 @@
  *  Created on: Nov 28, 2010
  *      Author: esban
  */
-#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
diff -Naur zipgateway/usr/local/contiki/cpu/native/net/tapdev6.c zipgateway_mod/usr/local/contiki/cpu/native/net/tapdev6.c
--- zipgateway/usr/local/contiki/cpu/native/net/tapdev6.c	2016-08-15 18:27:23.000000000 +0530
+++ zipgateway_mod/usr/local/contiki/cpu/native/net/tapdev6.c	2017-01-02 15:34:21.601911285 +0530
@@ -34,7 +34,7 @@
  * $Id: tapdev6.c,v 1.4 2010/10/19 18:29:04 adamdunkels Exp $
  */
 
-#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #include <fcntl.h>
 #include <stdlib.h>
 #include <stdio.h>
diff -Naur zipgateway/usr/local/contiki/platform/linux/parse_config.c zipgateway_mod/usr/local/contiki/platform/linux/parse_config.c
--- zipgateway/usr/local/contiki/platform/linux/parse_config.c	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/contiki/platform/linux/parse_config.c	2017-01-02 15:34:59.901913368 +0530
@@ -7,7 +7,7 @@
 #include "lib/list.h"
 
 /*For strdup! */
-#define _SVID_SOURCE
+#define _DEFAULT_SOURCE
 
 #include <string.h>
 #include <stdio.h>
diff -Naur zipgateway/usr/local/files/zipgateway.cfg zipgateway_mod/usr/local/files/zipgateway.cfg
--- zipgateway/usr/local/files/zipgateway.cfg	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/files/zipgateway.cfg	2017-01-02 15:36:22.861917881 +0530
@@ -1,17 +1,17 @@
 # Refer to zipgateway manual page in the documentation for
 #  more information.
 
-ZipSerialAPIPortName = /dev/ttyACM0
+ZipSerialAPIPortName = /dev/ttyS1
 ZipUnsolicitedDestinationIp6 = fd00:aaaa::1234
 ZipUnsolicitedDestinationPort = 41230
 #ZipUnsolicitedDestination2Ip6 = fd00:aaaa::1234
 #ZipUnsolicitedDestination2Port = 41231
 #SerialLog = /tmp/ziprouter.serlog
-ZipCaCert=/usr/local/etc/Portal.ca_x509.pem
-ZipCert=/usr/local/etc/ZIPR.x509_1024.pem
-ZipPrivKey=/usr/local/etc/ZIPR.key_1024.pem
+ZipCaCert=/etc/zipgateway/Portal.ca_x509.pem
+ZipCert=/etc/zipgateway/ZIPR.x509_1024.pem
+ZipPrivKey=/etc/zipgateway/ZIPR.key_1024.pem
 #Eepromfile=/etc/eeprom.dat
-#TunScript=/usr/local/etc/zipgateway.tun
+#TunScript=/etc/zipgateway/zipgateway.tun
 ZipPanIp6 = fd00:bbbb::1
 ZipLanIp6 = fd00:aaaa::3
 ZipLanGw6 = fd00:aaaa::1234
@@ -26,5 +26,5 @@
 #ZipMBPort=41230
 #ZipMBDestinationIp6=
 #ZipMBMode=1
-ZipPSK=123456789012345678901234567890AA
+ZipPSK=12345678901234567890123456789012
 #ExtraClasses= 0x43 0x75
diff -Naur zipgateway/usr/local/main.conf zipgateway_mod/usr/local/main.conf
--- zipgateway/usr/local/main.conf	1970-01-01 05:30:00.000000000 +0530
+++ zipgateway_mod/usr/local/main.conf	2017-01-02 15:38:18.925924194 +0530
@@ -0,0 +1,2 @@
+[General]
+NetworkInterfaceBlacklist=vmnet,vboxnet,virbr,ifb,ve-,br-lan,tap0
diff -Naur zipgateway/usr/local/src/serialapi/Serialapi.c zipgateway_mod/usr/local/src/serialapi/Serialapi.c
--- zipgateway/usr/local/src/serialapi/Serialapi.c	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/src/serialapi/Serialapi.c	2017-01-02 15:39:07.193926820 +0530
@@ -80,7 +80,7 @@
 //For Sleep call
 #include <windows.h>
 #else
-#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #include <unistd.h>
 #endif
 
diff -Naur zipgateway/usr/local/src/zwpgrmr/linux_usb_interface.c zipgateway_mod/usr/local/src/zwpgrmr/linux_usb_interface.c
--- zipgateway/usr/local/src/zwpgrmr/linux_usb_interface.c	2016-08-25 13:39:31.000000000 +0530
+++ zipgateway_mod/usr/local/src/zwpgrmr/linux_usb_interface.c	2017-01-02 15:39:35.721928372 +0530
@@ -80,7 +80,7 @@
  *  Created on: Nov 28, 2010
  *      Author: esban
  */
-#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
diff -Naur zipgateway/usr/local/zipgateway.cfg zipgateway_mod/usr/local/zipgateway.cfg
--- zipgateway/usr/local/zipgateway.cfg	1970-01-01 05:30:00.000000000 +0530
+++ zipgateway_mod/usr/local/zipgateway.cfg	2017-01-02 15:40:34.305931558 +0530
@@ -0,0 +1,13 @@
+ZipPanIp6 = fd00:bbbb::1
+ZipPanIp6PrefixLength = 64
+ZipUnsolicitedDestinationIp6 = fd00:aaaa::1234
+ZipUnsolicitedDestinationPort = 41230
+ZipLanIp6 = fd00:aaaa::3
+ZipLanIp6PrefixLength = 64
+ZipLanGw6 = fd00:aaaa::1234
+ZipSerialAPIPortName=/dev/ttyS1
+ZipCaCert=/etc/zipgateway/Portal.ca_x509.pem
+ZipCert=/etc/zipgateway/ZIPR.x509_1024.pem
+ZipPrivKey=/etc/zipgateway/ZIPR.key_1024.pem
+ZipPSK=12345678901234567890123456789012
+
diff -Naur zipgateway/usr/local/files/zipgateway.tun zipgateway_mod/usr/local/files/zipgateway.tun
--- zipgateway/usr/local/files/zipgateway.tun     2018-03-26 11:37:48.723893688 +0800
+++ zipgateway_mod/usr/local/files/zipgateway.tun      2018-03-26 11:38:09.455931141 +0800
@@ -30,7 +30,7 @@
            ifconfig $TUNDEV promisc
            sysctl -w net.ipv4.ip_forward=1
            sysctl -w net.ipv6.conf.$TUNDEV.accept_ra=2
-           TAP0=$(grep "^[[:space:]]*ZipLanGw6*" /usr/local/etc/zipgateway.cfg |awk '{ gsub (" ", "", $0); print}'|sed 's/ZipLanGw6=//')
+           TAP0=$(grep "^[[:space:]]*ZipLanGw6*" /etc/zipgateway/zipgateway.cfg |awk '{ gsub (" ", "", $0); print}'|sed 's/ZipLanGw6=//')
            ip -6 addr add $TAP0 dev $TUNDEV
            pidof parprouted > /dev/null 2>&1 || {
                /usr/sbin/parprouted $WLAN $TUNDEV

