diff -Naur zipctl/src/zwave/hcapi/demos/add_node/Makefile zipctl_mod/src/zwave/hcapi/demos/add_node/Makefile
--- a/zipctl/src/zwave/hcapi/demos/add_node/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/add_node/Makefile	2016-12-08 11:20:08.496281897 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/basic/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/basic/Makefile
--- a/zipctl/src/zwave/hcapi/demos/basic/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/basic/Makefile	2016-12-08 11:20:29.048281529 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/bin_switch/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/bin_switch/Makefile
--- a/zipctl/src/zwave/hcapi/demos/bin_switch/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/bin_switch/Makefile	2016-12-08 11:20:52.224281113 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/gw_discovery/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/gw_discovery/Makefile
--- a/zipctl/src/zwave/hcapi/demos/gw_discovery/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/gw_discovery/Makefile	2016-12-08 11:21:12.004280759 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/nw_reset/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/nw_reset/Makefile
--- a/zipctl/src/zwave/hcapi/demos/nw_reset/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/nw_reset/Makefile	2016-12-08 11:21:27.572280480 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/rm_node/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/rm_node/Makefile
--- a/zipctl/src/zwave/hcapi/demos/rm_node/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/rm_node/Makefile	2016-12-08 11:21:47.976280115 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf
 
diff -Naur zipctl/src/zwave/hcapi/demos/bin_sensor/Makefile zipctl-sigma_mod/src/zwave/hcapi/demos/bin_sensor/Makefile
--- a/zipctl/src/zwave/hcapi/demos/bin_sensor/Makefile	2015-09-04 13:58:39.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/demos/bin_sensor/Makefile	2016-12-08 11:21:47.976280115 +0530
@@ -2,8 +2,8 @@
 # Makefile to make sample application
 ################################################################################
 
-CC=gcc
-CFLAGS=
+#CC=gcc
+CFLAGS=-m32
 LDFLAGS=
 
 RM := rm -rf

diff -Naur zipctl/src/zwave/hcapi/lib/Makefile zipctl-sigma_mod/src/zwave/hcapi/lib/Makefile
--- a/zipctl/src/zwave/hcapi/lib/Makefile	2015-02-13 09:08:18.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/lib/Makefile	2016-12-08 11:18:17.336283887 +0530
@@ -2,8 +2,8 @@
 # Makefile to make zip api library
 ################################################################################
 
-CC=gcc
-AR=ar
+#CC=gcc
+#AR=ar
 CFLAGS=
 
 
diff -Naur zipctl/src/zwave/hcapi/src/Makefile zipctl-sigma_mod/src/zwave/hcapi/src/Makefile
--- a/zipctl/src/zwave/hcapi/src/Makefile	2016-09-16 09:29:36.000000000 +0530
+++ b/zipctl/src/zwave/hcapi/src/Makefile	2016-12-08 12:00:30.428238524 +0530
@@ -2,8 +2,8 @@
 # Makefile to make zipctl library
 ################################################################################
 
-CC=gcc
-AR=ar
+#CC=gcc
+#AR=ar
 CFLAGS=
 
 RM := rm -rf
