--- lua-openssl-0.5.1/src/xalgor.c	2015-12-15 14:04:23.000000000 +0800
+++ lua-openssl-0.5.1/src/xalgor.c	2016-07-19 10:26:08.421690606 +0800
@@ -34,7 +34,7 @@
   return 1;
 }
 
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+#if OPENSSL_VERSION_NUMBER >= 0x10003000L
 static int openssl_xalgor_cmp(lua_State* L) {
   X509_ALGOR* alg = CHECK_OBJECT(1, X509_ALGOR, "openssl.x509_algor");
   X509_ALGOR* ano = CHECK_OBJECT(2, X509_ALGOR, "openssl.x509_algor");
@@ -108,7 +108,7 @@
   {"md",                openssl_xalgor_md},
 #endif
   {"tostring",          openssl_xalgor_tostring},
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+#if OPENSSL_VERSION_NUMBER >= 0x10003000L
   {"equals",            openssl_xalgor_cmp},
   {"__eq",              openssl_xalgor_cmp},
 #endif
